grease/src/Grease/Concretize.hs

```diff
@@ -133,17 +134,29 @@ data ConcretizedData sym ext argTys
   -- ^ Concretized values from the @GlobalVar 'ToConcretizeType'@
   , concFs :: ConcFs
   , concMem :: ConcMem sym
-  , concErr :: Maybe (Mem.BadBehavior sym)
+  , concErr :: Maybe (ErrorDescription sym)
   }
 
+class Concretize e where
+  concretize :: forall sym t st fs. (WI.IsExprBuilder sym, sym ~ W4.ExprBuilder t st fs) => sym -> W4.GroundEvalFn t -> e sym -> IO (e sym)
+
+instance Concretize ErrorDescription where
+  concretize sym (W4.GroundEvalFn gFn) (CrucibleLLVMError bb cs) = do
+    bb' <- Mem.concBadBehavior sym gFn bb
+    pure (CrucibleLLVMError bb' cs)
+  concretize sym (W4.GroundEvalFn gFn) (MacawMemError memerr) = do
+    (UnmappedGlobalMemoryAccess ptrVal) <- pure memerr
+    cptr <- Mem.concPtr sym gFn ptrVal
+    pure $ MacawMemError (UnmappedGlobalMemoryAccess cptr)
```

Comment:
Why a typeclass instead of a simple function `concretizeErrorDescription`? Such a function could live in [ErrorDescription.hs](https://github.com/GaloisInc/grease/pull/284/files#diff-1a24746c2d30c235e89c047b70d5f63ed6b4d26bc185ca1061548650d1d5fdf7)